ycmd/responses.py

```diff
@@ -110,30 +110,51 @@ def BuildCompletionResponse( completion_datas, start_column ):
     'completion_start_column': start_column
   }
 
+def BuildLocationData( location ):
+  return {
+    'line_num': location.line_number_,
+    'column_num': location.column_number_,
+    'filepath': location.filename_,
+  }
 
-def BuildDiagnosticData( diagnostic ):
-  def BuildRangeData( source_range ):
-    return {
-      'start': BuildLocationData( source_range.start_ ),
-      'end': BuildLocationData( source_range.end_ ),
-    }
+def BuildRangeData( source_range ):
+  return {
+    'start': BuildLocationData( source_range.start_ ),
+    'end': BuildLocationData( source_range.end_ ),
+  }
 
-  def BuildLocationData( location ):
-    return {
-      'line_num': location.line_number_,
-      'column_num': location.column_number_,
-      'filepath': location.filename_,
-    }
+def BuildDiagnosticData( diagnostic ):
 
   kind = ( diagnostic.kind_.name if hasattr( diagnostic.kind_, 'name' )
            else diagnostic.kind_ )
 
+  fixits = ( diagnostic.fixits_ if hasattr( diagnostic, 'fixits_' )
+                              else [] )
```

Comment:
The indentation seems weird here a bit. Maybe put the `if` and what follows it on all on one separate line?